WalletWasabi/ExchangeRate/ExchangeRateProvider.cs

```diff
@@ -0,0 +1,79 @@
+using System.Collections.Immutable;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+using WalletWasabi.Extensions;
+using ExchangeRateProviderInfo = (string Name, string ApiUrl, bool ClearNetOnly, System.Func<string, decimal> Extractor);
+
+namespace WalletWasabi.ExchangeRate;
+
+public class ExchangeRateProvider(EndPoint? socksProxyEndPoint = null)
+{
+	private static readonly string ProviderList = @"""
+	Name       ClearNetOnly	 ApiUrl	                                                                    JsonPath
+    ---------  ------------  -------------------------------------------------------------------------  ------------------
+	bitstamp   true          https://www.bitstamp.net/api/v2/ticker/btcusd                              .bid
+	blockchain false         https://blockchain.info/ticker                                             .USD.buy
+	coinbase   true          https://api.coinbase.com/v2/exchange-rates?currency=BTC                    .data.rates.USD
```

Comment:
I was able to get this by using Tor Browser.

Reply:
No, that's not possible. I have verified again and that's not correct.

Reply:
![image](https://github.com/zkSNACKs/WalletWasabi/assets/9844978/d7d96c25-bee6-4e9f-8c00-4a1071e25a84)